### ISSUE TITLE
[feat] Disabled opcodes (CVE-2010-5137)

### ISSUE THREAD

**Issue by 0xBEEFCAF3** [keep-starknet-strange/shinigami#56]
Several opcodes have been disabled by the infamous 2010 commit [misc. changes](https://github.com/bitcoin/bitcoin/commit/4bd188c4383d6e614e18f79dc337fbabe8464c82#diff-27496895958ca30c47bbb873299a2ad7a7ea1003a9faa96b317250e3b7aa1fefR94).

#### References
https://github.com/bitcoin/bitcoin/blob/c9b7a792e2d8cfe5be14829ee6108d24abf8fcf6/src/script/interpreter.cpp#L456

[Guide on how to add an opcode in shinigami](https://github.com/keep-starknet-strange/shinigami/blob/main/resources/how-to-add-an-opcode.md)

**Comment by 0xBEEFCAF3** [comment]
I'd like to hop on this myself.

**Comment by b-j-roberts** [comment]
Hey @0xBEEFCAF3 , would love to have you on the project! I'll assign this to you now! Feel free to let me know if you have any questions.

Do you intend to implement these opcodes, or just the logic used to mark the disabled opcodes?

**Comment by 0xBEEFCAF3** [comment]
> d to mark the disabled op

@b-j-roberts I was planning on recreating the functionality that is in Bitcoin mainnet today. i.e the script should automatically fail.

**Comment by b-j-roberts** [comment]
@0xBEEFCAF3 Our primary goal is to create a VM compatible with mainnet, so I think this is what we will need. Longer term, I'd like to add some configuration flags to enable/disable certain opcodes, like OP_CAT, for example, so if you'd like to keep that in mind as part of the design of this work, that'd be cool. If not, disabling initially is a good start.

**Comment by 0xBEEFCAF3** [comment]
Yeah that makes sense. I'll start with just disabling as I am a cairo noob. Then it would make sense the re-create the same script flags as bitcoin-core which then allow certain OP_SUCCESS / OP_NOP opcodes to be replaced with newer functionality. This is how we re-enabled OP_CAT in Bitcoin signet.
https://github.com/bitcoin-inquisition/bitcoin/pull/39
